src/app/dataset-view/additional-components/flows-component/flows.component.ts

```diff
@@ -0,0 +1,162 @@
+        this.onPageChangeEmit.emit(page);
+    }
+
+    public datasetFlowListByPage(): void {
+        const pageParam = this.activatedRoute.snapshot.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_PAGE);
+        if (pageParam) {
+            this.currentPage = +requireValue(pageParam);
+            this.getFlowConnectionData(this.currentPage, this.filterByStatus);
+        } else {
+            this.currentPage = 1;
+            this.getFlowConnectionData(this.currentPage, this.filterByStatus);
+        }
+    }
+
+    public updateSettings(): void {
+        this.navigationService.navigateToDatasetView({
+            accountName: this.datasetBasics.owner.accountName,
+            datasetName: this.datasetBasics.name,
+            tab: DatasetViewTypeEnum.Settings,
+            section: SettingsTabsEnum.SCHEDULING,
+        });
+    }
+
+    public updateNow(): void {
+        this.getTileWidgetData();
+        this.getFlowConnectionData(this.currentPage, this.filterByStatus);
+    }
+
+    public onChangeFilterByStatus(status: MaybeNull<FlowStatus>): void {
+        this.getFlowConnectionData(this.currentPage, status);
+        this.filterByStatus = status;
+    }
+
+    public onChangeFilterByInitiator(initiator: string): void {
+        if (initiator !== "Account") {
+            let filterOptions: MaybeNull<InitiatorFilterInput> = null;
+            if (initiator === "System") {
+                filterOptions = { system: true };
+            }
+            this.getFlowConnectionData(this.currentPage, this.filterByStatus, filterOptions);
+        }
+        this.filterByInitiator = initiator;
+    }
+
+    public onSearchByAccountName(accountName: string): void {
+        this.getFlowConnectionData(this.currentPage, this.filterByStatus, { account: accountName });
+        this.searchByAccountName = accountName;
+    }
+
+    public pauseDatasetUpdates(flow: FlowDataFragment): void {
+        if (flow.status !== FlowStatus.Finished) {
+            this.trackSubscription(
+                this.flowsService
+                    .datasetPauseFlows({
+                        datasetId: this.datasetBasics.id,
+                        datasetFlowType:
+                            this.datasetBasics.kind === DatasetKind.Root
+                                ? DatasetFlowType.Ingest
+                                : DatasetFlowType.ExecuteTransform,
+                    })
+                    .subscribe(),
+            );
+        } else {
+            this.trackSubscription(
+                this.flowsService
+                    .datasetResumeFlows({
+                        datasetId: this.datasetBasics.id,
+                        datasetFlowType:
+                            this.datasetBasics.kind === DatasetKind.Root
+                                ? DatasetFlowType.Ingest
+                                : DatasetFlowType.ExecuteTransform,
+                    })
+                    .subscribe(),
+            );
+        }
+    }
+}
```

Comment:
I don't understand the logic.

1) If flow is not finished, we allow pausing a configuration.
If flow is finished, we allow resuming - but what if there is another flow running?
Resuming should be allowed only when the configuration is PAUSED, not depending oon the flow sstatus.

2) What are you going to do with other types of flows. We are going to implement COMPACTION flow in few weeks.

3) The method is probably named wrong. It's something that toggles pausing/resuming all dataset flows, not just 1 configuration like "INGEST".